docs/specs/user-model.md

```diff
@@ -148,6 +148,12 @@ Login ID type determines the validation, normalization and unique key generation
 
 - [RFC5322](https://tools.ietf.org/html/rfc5322#section-3.4.1)
 - Disallow `+` sign in the local part (Configurable, default OFF)
+- Domain blacklist / whitelist
+  - Block blacklisted domains (Configurable, default OFF, can be ON only if *Allow whitelisted domains* is OFF)
+  - Include default blacklisted domains (Configurable, default OFF, can be ON only if *Block blacklisted domains* is ON)
+  - Allow whitelisted domains only (Configurable, default OFF, can be ON only if *Block blacklisted domains* is OFF)
+  - Domain blacklist / whitelist only affect user signup, users created from portal or via admin API are not affected
+  - Developer will need to provide their blacklist / whitelist in txt file, separated by newline. A default blacklist will be included when *Include default blacklisted domains* is ON
```

Comment:
Seems inconsistent with keywords in username? Shall we stick with one?

Reply:
Just afraid that the list will be too long to save in config, maybe we can:

1. If we think the developer provided list will not be too long, we can move the blacklist / whitelist config back to the yaml config file. So it will be consistent to username's excluded keywords.

or

2. Move username's excluded keywords to txt file. But this will be another breaking change..>"<

Thought?